osu.Game/Configuration/SettingSourceAttribute.cs

```diff
@@ -139,6 +150,30 @@ public static IEnumerable<Drawable> CreateSettingsControls(this object obj)
             }
         }
 
+        public static IEnumerable<(SettingSourceAttribute, PropertyInfo)> GetSettingsSourcePropertiesFromBindables(this object obj)
+        {
+            HashSet<string> handledProperties = new HashSet<string>();
+
+            // reverse and de-dupe properties to surface base class settings to the top of return order.
```

Comment:
Why? Shouldn't the most derived class always store the most concrete implementation?

Reply:
This is so that `InputHandler` can display things like `Enabled` before the more local settings of the derived classes.